packages/zoe/src/zoeService/zoe.js

```diff
@@ -243,6 +218,68 @@ function makeZoe(vatAdminSvc, zcfBundleName = undefined) {
             zoeSeatAdmins.forEach(zoeSeatAdmin => zoeSeatAdmin.fail(reason));
           },
           stopAcceptingOffers: () => (acceptingOffers = false),
+          makeUserSeat: async (
+            invitationHandle,
+            initialAllocation,
+            proposal,
+          ) => {
+            const offerResultPromiseKit = makePromiseKit();
+            // Don't trigger Node.js's UnhandledPromiseRejectionWarning.
+            // This does not suppress any error messages.
+            offerResultPromiseKit.promise.catch(_ => {});
+            const exitObjPromiseKit = makePromiseKit();
+            // Don't trigger Node.js's UnhandledPromiseRejectionWarning.
+            // This does not suppress any error messages.
+            exitObjPromiseKit.promise.catch(_ => {});
+            const seatHandle = makeHandle('SeatHandle');
+
+            const { userSeat, notifier, zoeSeatAdmin } = makeZoeSeatAdminKit(
+              initialAllocation,
+              exitZoeSeatAdmin,
+              hasExited,
+              proposal,
+              withdrawPayments,
+              exitObjPromiseKit.promise,
+              offerResultPromiseKit.promise,
+            );
+
+            seatHandleToZoeSeatAdmin.init(seatHandle, zoeSeatAdmin);
+
+            const seatData = harden({ proposal, initialAllocation, notifier });
+
+            zoeSeatAdmins.add(zoeSeatAdmin);
+
+            E(/** @type {Promise<AddSeatObj>} */ (addSeatObjPromiseKit.promise))
+              .addSeat(invitationHandle, zoeSeatAdmin, seatData, seatHandle)
+              .then(({ offerResultP, exitObj }) => {
```

Comment:
Just noting that this `then` is perfectly safe because the chain starts in an `E`. When we eventually check for inappropriate `then`s, we will need to consider this pattern appropriate. No change suggested.